mypyc/ops_primitive.py

```diff
@@ -153,9 +153,20 @@ def name_ref_op(name: str,
 def custom_op(arg_types: List[RType],
               result_type: RType,
               error_kind: int,
-              format_str: str,
               emit: EmitCallback,
+              name: Optional[str] = None,
+              format_str: Optional[str] = None,
               is_var_arg: bool = False) -> OpDescription:
     """Create a one-off op that can't be automatically generated from the AST."""
+    if name is not None:
+        typename = ''
+        if len(arg_types) == 1:
+            typename = ' :: %s' % short_name(arg_types[0].name)
+        if format_str is None:
+            format_str = '{dest} = %s %s%s' % (short_name(name),
+                                           ', '.join('{args[%d]}' % i
+                                                     for i in range(len(arg_types))),
+                                           typename)
+    assert format_str is not None
```

Comment:
Maybe add a more informative message about what exactly is required (either `name` or `format_str` to be non-None)?

Reply:
I meant to the assert, but the documentation is good too :)